test/iterable_extensions_test.dart

```diff
@@ -32,6 +32,46 @@ void main() {
     expect(actual, equals([1, 2, 'a', 'b']));
   });
 
+  group('partition', () {
```

Comment:
Could you add a test for where the list has less than n items in it. I have a feeling there is a bug.

i.e. `[0, 1].partition(3)`